src/test/kotlin/uk/gov/dluhc/printapi/database/repository/AnonymousElectorDocumentSummaryRepositoryIntegrationTest.kt

```diff
@@ -0,0 +1,222 @@
+
+import org.assertj.core.api.Assertions.assertThat
+import org.junit.jupiter.api.Nested
+import org.junit.jupiter.api.Test
+import org.springframework.data.domain.PageRequest
+import org.springframework.data.domain.Sort
+import uk.gov.dluhc.printapi.config.IntegrationTest
+import uk.gov.dluhc.printapi.database.entity.SourceType.ANONYMOUS_ELECTOR_DOCUMENT
+import uk.gov.dluhc.printapi.testsupport.testdata.aGssCode
+import uk.gov.dluhc.printapi.testsupport.testdata.aValidApplicationReference
+import uk.gov.dluhc.printapi.testsupport.testdata.aValidSourceReference
+import uk.gov.dluhc.printapi.testsupport.testdata.anotherGssCode
+import uk.gov.dluhc.printapi.testsupport.testdata.entity.buildAedContactDetails
+import uk.gov.dluhc.printapi.testsupport.testdata.entity.buildAnonymousElectorDocument
+import uk.gov.dluhc.printapi.testsupport.testdata.entity.buildAnonymousElectorDocumentSummaryViewFromAedEntity
+import java.time.Instant
+import java.time.LocalDate
+
+internal class AnonymousElectorDocumentSummaryRepositoryIntegrationTest : IntegrationTest() {
+
+    @Nested
+    inner class FindByGssCodeInAndSourceTypeOrderByDateCreatedDescSanitizedSurnameAsc {
+
+        @Test
+        fun `should returns empty list when summary view has no records`() {
+            // Given
+            val gssCode = aGssCode()
+
+            // When
+            val actual = anonymousElectorDocumentSummaryRepository
+                .findAllByGssCodeInAndSourceType(
+                    gssCodes = listOf(gssCode),
+                    sourceType = ANONYMOUS_ELECTOR_DOCUMENT,
+                    pageRequest = withPageRequestAndSortOrder(0, 100)
+                )
+
+            // Then
+            assertThat(actual).isNotNull.isEmpty()
+        }
+
+        @Test
+        fun `should returns empty list when no matching gssCode record exists`() {
+            // Given
+            val gssCode = aGssCode()
+            val otherAed = buildAnonymousElectorDocument(gssCode = anotherGssCode())
+            anonymousElectorDocumentRepository.saveAll(listOf(otherAed))
+
+            // When
+            val actual = anonymousElectorDocumentSummaryRepository
+                .findAllByGssCodeInAndSourceType(
+                    gssCodes = listOf(gssCode),
+                    sourceType = ANONYMOUS_ELECTOR_DOCUMENT,
+                    pageRequest = withPageRequestAndSortOrder(0, 100)
+                )
+
+            // Then
+            assertThat(actual).isNotNull.isEmpty()
+        }
+
+        @Test
+        fun `should find all AEDs for a given gssCode and source type pages`() {
+            // Given
+            val gssCode = aGssCode()
+            val currentDate = LocalDate.now()
+            val currentDateTimeInstant = Instant.now()
+            val aed1SourceReference = aValidSourceReference()
+            val aed1ApplicationReference = aValidApplicationReference()
+
+            val aed1FirstCertificate = buildAnonymousElectorDocument(
```

Comment:
Certificate in the names? I'm not sure I understand the names of these variables. Also, this is a large given section, which makes it difficult to grasp the intention of the test.